var/spack/repos/builtin/packages/symengine/package.py

```diff
@@ -0,0 +1,118 @@
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Symengine(Package):
+    """SymEngine is a fast symbolic manipulation library, written in C++."""
+
+    homepage = "http://sympy.org"
+    url      = "https://github.com/symengine/symengine/archive/v0.2.0.tar.gz"
+
+    version('0.2.0', '45401561add36a13c1f0b0c5f8d7422d')
+    version('0.1.0', '41ad7daed61fc5a77c285eb6c7303425')
+    version('develop', git='https://github.com/symengine/symengine.git')
+
+    # Libraries to add to spack:
+    # Flint: www.flintlib.org
+
+    # variant('bfd',          default=False,
+    #         description='Compile with BFD library')  # Error: Requires linkh?
+    # variant('flint',        default=False,
+    #         description='Compile with Flint library')
+    variant('mpc',          default=True,
+            description='Compile with MPC library')
+    variant('mpfr',         default=True,
+            description='Compile with MPFR library')
+    variant('piranha',      default=False,
+            description='Compile with Piranha library')
+    variant('openmp',       default=False,
+            description='Enable OpenMP support')
+    variant('shared',       default=True,
+            description='Enables the build of shared libraries')
+
+    # Build dependencies
+    depends_on('cmake',    type='build')
+
+    # Other dependencies
+    depends_on('gmp')
+    depends_on('binutils', when='+bfd')
+    depends_on('mpc',      when='+mpc')
+    depends_on('mpfr',     when='+mpfc')
+    depends_on('piranha',  when='+piranha')
+    # depends_on('flint',    when='+flint~piranha')
+
+    def install(self, spec, prefix):
+        options = []
+        options.extend(std_cmake_args)
+
+        # CMAKE_BUILD_TYPE should be Debug | Release
+        for word in options[:]:
+            if word.startswith('-DCMAKE_BUILD_TYPE'):
+                options.remove(word)
+
+        # See https://github.com/symengine/symengine/blob/master/README.md
+        # for build options
+        options.extend([
+            '-DCMAKE_BUILD_TYPE=Release',
+            '-DWITH_SYMENGINE_RCP:BOOL=ON',
+            '-DWITH_SYMENGINE_THREAD_SAFE:BOOL=ON',
```

Comment:
@isuruf Is this good to have set to `ON` by default? I expect to use it in a mpi/multithreading context in the future. I think that this is the only option from your list of defaults that I've changed.